### ISSUE TITLE
Add squiggles in c_cpp_properties.json for browse.databaseFilename if it is not the first folder in the multi-root workspace

### ISSUE THREAD

**Issue by xolom** [microsoft/vscode-cpptools#6838]
**Version info**
- OS and Version: MacOS 11.1
- VS Code Version: 1.52.1
- C/C++ Extension Version: 1.2.0-insiders2
- Other extensions you installed (and if the issue persists after disabling them): Tried without every other extension

**Bug description**

I have set `browse.databaseFilename` in my `c_cpp_properties.json`, unfortunately it looks like that property isn't used/applied. But i already found out why it isn't working.

Content of my `c_cpp_properties.json`:
```
{
    "configurations": [
        {
            "name": "Configuration 1",
            "defines": [],
            "cStandard": "c11",
            "cppStandard": "c++17",
            "intelliSenseMode": "clang-x64",
            "browse": {
                "databaseFilename": "/tmp/vscode/browse_config10.db"
            }
        }
    ],
    "version": 4
}
```

**Now here comes the thing:** I noticed that the order of workspace folders is affecting this behavior. Here is the content of my workspace json file:
```
{
	"folders": [
		{
			"path": "build"
		},
		{
			"path": "src"
		},
	],
	"settings": {
		"C_Cpp.loggingLevel": "Debug",
	}
}
```
This is my folder/file structue:
```
├── build
│   └── .vscode
│       └── settings.json
├── src
│   ├── .vscode
│   │   └── c_cpp_properties.json
│   └── main.cpp
└── workspace.code-workspace
```
**As soon as i put the src folder to the top** (where `c_cpp_properties.json` is located in), **it works as expected!**
```
"folders": [
	{
		"path": "src"
	},
	{
		"path": "build"
	}
],
```

Maybe this is intended and I'm not aware of, but i don't think so.
I attached a zipfile with my minimal example, so this should be easy to reproduce: [code_workspace.zip](https://github.com/microsoft/vscode-cpptools/files/5859942/code_workspace.zip)

**Comment by sean-mcmanus** [comment]
Yeah, this is currently "by design" behavior. The reason is because the databaseFilename is shared with all workspace folders so we just look at the 1st folder's setting. You could also use the C_Cpp.default.databaseFilename at a workspace or user level. The problem with looking for databaseFilename in other workspace folders is because each workspace folder could have a different one and then there'd a question of which one to choose.

We could use this issue to track potentially changing the design.
